scipy/signal/tests/test_signaltools.py

```diff
@@ -453,27 +699,107 @@ def test_valid_mode(self):
         out = fftconvolve(b, a, 'valid')
         assert_array_almost_equal(out, expected)
 
+        out = fftconvolve(a, b, 'valid', axes=0)
+        assert_array_almost_equal(out, expected)
+
+        out = fftconvolve(b, a, 'valid', axes=0)
+        assert_array_almost_equal(out, expected)
+
+        out = fftconvolve(a, b, 'valid', axes=[0])
+        assert_array_almost_equal(out, expected)
+
+        out = fftconvolve(b, a, 'valid', axes=[0])
+        assert_array_almost_equal(out, expected)
+
+        out = fftconvolve(a, b, 'valid', axes=-1)
+        assert_array_almost_equal(out, expected)
+
+        out = fftconvolve(b, a, 'valid', axes=-1)
+        assert_array_almost_equal(out, expected)
+
+        out = fftconvolve(a, b, 'valid', axes=[-1])
+        assert_array_almost_equal(out, expected)
+
+        out = fftconvolve(b, a, 'valid', axes=[-1])
+        assert_array_almost_equal(out, expected)
```

Comment:
simplify this with:
```
for axes in (0, [0], -1, [-1]):
     out = ...(a, b, ...
     assert...
     out = ...(b, a, ...
     assert...
```
Same, if possible, for other tests. It makes it obvious where the repetition is.

Reply:
I use `pytest.mark.parameterize`.  This has the benefit over a simple loop that it automatically tells you which values caused the error.  I also replaced an existing loop in the test with this.